updatecli/updatecli.d/trusted.ci.sp.endate.yaml

```
@@ -0,0 +1,62 @@
+name: "Generate new end date for trusted.ci.jenkins.io controller service principal"
+
+scms:
+  default:
+    kind: github
+    spec:
+      user: "{{ .github.user }}"
+      email: "{{ .github.email }}"
+      owner: "{{ .github.owner }}"
+      repository: "{{ .github.repository }}"
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
+      branch: "{{ .github.branch }}"
+
+sources:
+  currentExpiry:
+    name: Get current `expiry` date
+    kind: terraform/file #  as per updatecli/updatecli#1859 need to use terraform/file findsubmatch for now
+    spec:
+      file: trusted.ci.jenkins.io.tf
+      path: 'module.controller_service_principal_end_date'
+  nextExpiry:
+    name: Prepare next `expiry` date within 3 months
+    kind: shell
+    spec:
+      command: bash ./updatecli/scripts/dateadd.sh
+      environments:
+        - name: PATH
+
+conditions:
+  checkIfExpirySoonExpired:
+    kind: shell
+    sourceid: currentExpiry
+    spec:
+      command: bash ./updatecli/scripts/datediff.sh # current expiry date value passed as argument
+      environments:
+        - name: PATH
+
+targets:
+  updateNextExpiry:
+    name: Update Terraform file `trusted.ci.jenkins.io.tf` with new expiration date for controller_service_principal_end_date
+    kind: file
+    sourceid: nextExpiry
+    spec:
+      file: trusted.ci.jenkins.io.tf
+      matchpattern: (controller_service_principal_end_date.*=.*\")(.*)(\".*)
+      replacepattern: ${1}{{ source "nextExpiry" }}"
+    scmid: default
+
+actions:
+  default:
+    kind: github/pullrequest
+    scmid: default
+    spec:
+      title: Generate a new Azure Service Principal password with expiration date {{ source "nextExpiry" }} on `trusted.ci.jenkins.io`
```

```suggestion
      title: Generate a new Azure AD Application password with expiration date {{ source "nextExpiry" }} on `trusted.ci.jenkins.io`
```

It's an Azure AD password, not a Service Principal password as per https://github.com/jenkins-infra/shared-tools/blob/adae7d43808f77c2a94bac3846d6219da93b3261/terraform/modules/azure-jenkinsinfra-controller/main.tf#L303-L307